lint/checker_base.go

```diff
@@ -57,25 +61,37 @@ type (
 	//	- Every LHS of ":=" assignment
 	//	- Every local var/const declaration.
 	//
-	// See also: baseLocalNameChecker, wrapLocalNameChecker.
+	// See also: wrapLocalNameChecker.
 	localNameChecker interface {
+		walkerEventHandler
 		CheckLocalName(*ast.Ident)
 	}
 
 	// typeExpeChecker visits every type describing expression.
 	// It also traverses struct types and interface types to run
 	// checker over their fields/method signatures.
 	//
-	// See also: baseTypeExprChecker, wrapTypeExprChecker.
+	// See also: wrapTypeExprChecker.
 	typeExprChecker interface {
+		walkerEventHandler
 		CheckTypeExpr(ast.Expr)
 	}
 )
 
-type baseFuncDeclChecker struct {
+type checkerBase struct {
 	ctx *context
 }
 
+func (c *checkerBase) BindContext(ctx *context) {
+	c.ctx = ctx
+}
+
+func (c *checkerBase) Init() {}
```

Comment:
What about a comment that states that `Init` should be implemented in checker to initialise some data?

Reply:
Also commented on BindContext and added a "See checkerBase" for abstractChecker type comment.